File: src/main/scala/com/vc/easy/L824.scala
package com.vc.easy

object L824 {
  def toGoatLatin(S: String): String = {
    import scala.collection.mutable
    val vowel = mutable.HashSet('a','e','i','o','u','A','E','I','O','U')
    val str = new mutable.StringBuilder()
    var i = 0
    var j = 0
    var k = 0
    var count = 0
    while(j < S.length) {
      i = j
      while(j < S.length && S(j) != ' ') j += 1

      count += 1
      if(vowel.contains(S(i))) {
        str.append(S.substring(i, j))
        str.append("ma")
      }
      else {
        str.append(S.substring(i + 1, j))
        str.append(S(i))
        str.append("ma")
      }

      k = 0
      while(k < count) {
        str.append("a")
        k += 1
      }
      j += 1
      if(j < S.length) str.append(" ")
    }
    str.toString
  }
}

File: src/main/java/com/vc/easy/L824.java
package com.vc.easy;

import java.util.HashSet;

class L824 {
    public String toGoatLatin(String S) {
        HashSet<Character> vowels = new HashSet<>();
        vowels.add('a');
        vowels.add('e');
        vowels.add('i');
        vowels.add('o');
        vowels.add('u');

        StringBuilder sb = new StringBuilder();
        String[] splits = S.split(" ");
        StringBuilder suffix = new StringBuilder("a");

        for(String split: splits) {
            char ch = split.charAt(0);

            if(vowels.contains(Character.toLowerCase(ch))) {
                sb.append(split);
                sb.append("ma");
                sb.append(suffix.toString());
            }
            else {
                sb.append(split.substring(1));
                sb.append(split.charAt(0));
                sb.append("ma");
                sb.append(suffix.toString());
            }
            sb.append(" ");
            suffix.append("a");
        }
        sb.deleteCharAt(sb.length() - 1);
        return sb.toString();
    }
}
